leshan-server-core/src/main/java/org/eclipse/leshan/server/security/SecurityInfo.java

```diff
@@ -114,26 +121,42 @@ public static SecurityInfo newX509CertInfo(String endpoint) {
     /**
      * Construct a {@link SecurityInfo} when using OSCORE.
      */
-    public static SecurityInfo newOSCoreInfo(String endpoint, OSCoreCtx oscoreCtx) {
-        Validate.notNull(oscoreCtx);
+    // TODO OSCORE rename in newOscoreInfo
+    public static SecurityInfo newOSCoreInfo(String endpoint, OscoreSetting oscoreSetting) {
+        Validate.notNull(oscoreSetting);
 
+        // TODO OSCORE remove access to context here.
         // Add the OSCORE Context to the context database
         HashMapCtxDB db = OscoreServerHandler.getContextDB();
-        db.addContext(oscoreCtx);
+        db.addContext(getContext(oscoreSetting));
 
-        return new SecurityInfo(endpoint, null, null, null, false, oscoreCtx);
+        return new SecurityInfo(endpoint, null, null, null, false, oscoreSetting);
+    }
+
+    private static OSCoreCtx getContext(OscoreSetting oscoreSetting) {
+        try {
+            OSCoreCtx osCoreCtx = new OSCoreCtx(oscoreSetting.getMasterSecret(), true,
```

Comment:
I'm not sure to understand the second argument (boolean) ?
The javadoc says: 
```
 * @param client is this originally the client's context
```
But still not clear to me, I don't know if I should use true or false. (with true or false integration tests seems to be happy :thinking:  )

Reply:
According to the constructor of OSCoreCtx the ```boolean client``` only matters in the case if sender_id or recipient_id is not defined.

Reply:
Yes, indeed. That parameter is only relevant if the sender ID and recipient ID is not defined. I think the original intention was for an application the be able to create a very simple OSCORE context for testing.